common/src/main/java/io/netty/util/internal/StringUtil.java

```diff
@@ -314,6 +322,67 @@ public static String simpleClassName(Class<?> clazz) {
         }
     }
 
+    /**
+     * Escapes the specified value, if necessary according to
+     * <a href="https://tools.ietf.org/html/rfc4180#section-2">RFC-4180</a>.
+     * <p>
+     * The {@link CharSequence} will be escaped is if contains a
+     * <ul>
+     *     <li>{@link StringUtil#COMMA}</li>
+     *     <li>{@link StringUtil#LINE_FEED}</li>
+     *     <li>{@link StringUtil#CARRIAGE_RETURN}</li>
+     *     <li>{@link StringUtil#DOUBLE_QUOTE}</li>
+     * </ul>
+     * If a double quote is found inside of the value, it will be escaped by preceeding it
+     * with another double quote.
+     *
+     * @param value the value to be appended, and quoted if necessary
+     * @return {@link CharSequence} the quoted value
+     */
+    public static CharSequence escapeCsv(CharSequence value) {
+        int length = checkNotNull(value, "value").length();
+        if (length == 0) {
+            return value;
+        }
+        int last = length - 1;
+        boolean quoted = isDoubleQuote(value.charAt(0)) && isDoubleQuote(value.charAt(last)) && length != 1;
+        boolean foundSpecialCharacter = false;
+        boolean foundEscapedCharacter = false;
+        StringBuilder escaped = new StringBuilder(length).append(DOUBLE_QUOTE_STRING);
```

Comment:
At a minimum I think we need `length + 2`.  We may also want to add some other small delta to account for escape characters that may be added (this would be a guess).